eng/packages/http-client-csharp-mgmt/generator/Azure.Generator.Management/src/Providers/OperationMethodProviders/ResourceOperationMethodProvider.cs

```diff
@@ -224,6 +225,84 @@ private IReadOnlyList<MethodBodyStatement> BuildClientPipelineProcessing(
             }
         }
 
+        /// <summary>
+        /// Builds client pipeline handling with status code switch logic for exists/get-if-exists operations.
+        /// Handles 200 (success) and 404 (not found) status codes specifically.
+        /// </summary>
+        protected IReadOnlyList<MethodBodyStatement> BuildExistsOperationPipelineHandling(
+            VariableExpression messageVariable,
+            VariableExpression contextVariable,
+            out ScopedApi<Response> responseVariable)
+        {
+            var statements = new List<MethodBodyStatement>();
+
+            var sendMethod = _isAsync ? "SendAsync" : "Send";
+            var sendArguments = new ValueExpression[] { messageVariable, contextVariable.Property(nameof(RequestContext.CancellationToken)) };
+
+            var sendStatement = _isAsync
+                ? This.Property("Pipeline").Invoke(sendMethod, sendArguments, null, _isAsync).Terminate()
+                : This.Property("Pipeline").Invoke(sendMethod, sendArguments).Terminate();
+            statements.Add(sendStatement);
+
+            var resultDeclaration = Declare(
+                "result",
+                typeof(Response),
+                messageVariable.Property("Response"),
+                out var resultVariable);
+            statements.Add(resultDeclaration);
+
+            var responseDeclaration = Declare(
+                "response",
+                new CSharpType(typeof(Response<>), _originalBodyType!),
+                Default,
+                out var responseVar);
+            responseVariable = responseVar.As<Response>();
+            statements.Add(responseDeclaration);
+
+            var switchStatement = new SwitchStatement(resultVariable.Property("Status"));
+
+            // Helper method to create switch case body
+            static List<MethodBodyStatement> CreateSwitchCaseBody(VariableExpression responseVar, ValueExpression valueExpression, VariableExpression resultVariable)
```

Comment:
consider to put this inner method to the end of the method body for readability, and static can be removed

Reply:
put the method to the end of the method body as suggested but keep the static keyword as we did in other places in the library.